apps/cli/src/commands/build.ts

```diff
@@ -1,312 +1,645 @@
+    drive: EmptyDriveConfig,
+    sdkImage: string,
+    destination: string,
+): Promise<DriveResult> => {
+    const filename = `${name}.${drive.format}`;
+    switch (drive.format) {
+        case "ext2": {
+            const blockSize = 4096; // fixed at 4k
+            const size = Math.ceil(drive.size / blockSize); // size in blocks
+            const command = "xgenext2fs";
+            const args = [
+                "--block-size",
+                blockSize.toString(),
+                "--faketime",
+                "--size-in-blocks",
+                size.toString(),
+                filename,
+            ];
+            await execaDockerFallback(command, args, {
+                cwd: destination,
+                image: sdkImage,
+            });
+            break;
+        }
+        case "ext4": {
+            const blockSize = 4096; // fixed at 4k
+            const size = Math.ceil(drive.size / blockSize); // size in blocks
+            const seed = "00000000-0000-0000-0000-000000000001";
+            const command = "mke2fs";
+            const PATH = "/opt/homebrew/opt/e2fsprogs/sbin";
+            const args = [
+                "-b",
+                blockSize.toString(),
+                "-U",
+                "clear",
+                "-E",
+                `hash_seed=${seed}`,
+                "-t",
+                drive.format,
+                filename,
+                size.toString(),
+            ];
+            const env = { PATH, SOURCE_DATE_EPOCH: "0" };
+            await execaDockerFallback(command, args, {
+                cwd: destination,
+                env,
+                image: sdkImage,
+            });
+            break;
         }
 
-        // warn for using default values
-        info.sdkVersion ||
-            this.warn(
-                `Undefined ${CARTESI_LABEL_SDK_VERSION} label, defaulting to ${CARTESI_DEFAULT_SDK_VERSION}`,
+        case "raw": {
+            await fs.writeFile(
+                path.join(destination, filename),
+                Buffer.alloc(drive.size),
             );
+            break;
+        }
+    }
+    return { filename: path.join(destination, filename) };
+};
 
-        info.ramSize ||
-            this.warn(
-                `Undefined ${CARTESI_LABEL_RAM_SIZE} label, defaulting to ${CARTESI_DEFAULT_RAM_SIZE}`,
-            );
+const tarToExt = async (
+    input: string,
+    output: string,
+    format: "ext2" | "ext4",
+    extraSize: number,
+    options: OptionsDockerFallback,
+) => {
+    const blockSize = 4096; // fixed at 4k
+    const extraBlocks = Math.ceil(extraSize / blockSize);
+    // const extraSize = `+${extraBlocks}`;
+
+    const seed = "00000000-0000-0000-0000-000000000001";
+    const command = "mke2fs";
+    const PATH = "/opt/homebrew/opt/e2fsprogs/sbin";
+    const args = [
+        "-b",
+        blockSize.toString(),
+        "-U",
+        "clear",
+        "-E",
+        `hash_seed=${seed}`,
+        "-d",
+        input,
```

Comment:
Neither my local `mke2fs` (Ubuntu 24.04 `1.47.0 (5-Feb-2023)`) nor the one installed on cartesi/sdk accepts a tarball for `-d` argument.

So we'd need to unpack the tarball into a directory before at this `tarToExt()` function.

Or we should compile mke2fs inside cartesi/sdk with `-d` support to tarball, so that `execaDockerFallback()` doesn't fail right ahead.

Reply:
Only available on e2fsprogs >= 1.47.1

>   * Mke2fs -d can now support an input tar file if the libarchive library
    is available.

Reply:
That was why I changed the sdk to include version 1.47.1

Reply:
Yeah, and it was failing because this PR ignores the `LABEL`  inside `Docekrfile` and wasn't using the `cartesi/sdk:devel` but `0.10.0` version

it worked with `devel`, so I'm resolving this conversation